wasm/src/debug_service.rs

```diff
@@ -141,6 +201,64 @@ impl Default for DebugService {
     }
 }
 
+impl From<StepResult> for StructStepResult {
+    fn from(value: StepResult) -> Self {
+        match value {
+            StepResult::BreakpointHit(value) => StructStepResult {
+                id: StepResultId::BreakpointHit.into(),
+                value: Into::<usize>::into(value),
+            },
+            StepResult::Next => StructStepResult {
+                id: StepResultId::Next.into(),
+                value: 0,
+            },
+            StepResult::StepIn => StructStepResult {
+                id: StepResultId::StepIn.into(),
+                value: 0,
+            },
+            StepResult::StepOut => StructStepResult {
+                id: StepResultId::StepOut.into(),
+                value: 0,
+            },
+            StepResult::Return(_) => StructStepResult {
+                id: StepResultId::Return.into(),
+                value: 0,
+            },
+        }
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Copy, Clone, Serialize, Deserialize)]
+pub enum StepResultId {
```

Comment:
In general, we should avoid exporting enums at the wasm layer. There's no such thing in JavaScript, and TypeScript does it by exporting a value (a class that maps names to numbers) in the .d.ts, which then adds a bunch of footguns with importing platform specific JS in the platform agnostic code. Here I see you're only using this value in the platform specific 'browser.ts', so not currently an issue. A note for the future though, as we likely want guidance NOT to do this in general.